--- src/video/wayland/SDL_waylandevents.c	2023-03-12 11:34:57.705357013 +0000
+++ src/video/wayland/SDL_waylandevents.c	2023-03-12 11:35:02.320769000 +0000
@@ -1129,7 +1129,8 @@
         const SDL_Scancode scancode = Wayland_get_scancode_from_key(input, *key + 8);
 
         if (scancode != SDL_SCANCODE_UNKNOWN) {
-            for (uint32_t i = 0; i < sizeof mod_scancodes / sizeof *mod_scancodes; ++i) {
+            uint32_t i;
+            for (i = 0; i < sizeof mod_scancodes / sizeof *mod_scancodes; ++i) {
                 if (mod_scancodes[i] == scancode) {
                     SDL_SendKeyboardKey(SDL_PRESSED, scancode);
                     break;
